src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,276 @@
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/inlet.hpp"
+
+#include <iostream>
+
+#include "fmt/fmt.hpp"
+
+#include "axom/slic/core/SimpleLogger.hpp"
+
+namespace inlet = axom::inlet;
+using Vector = inlet::FunctionType::Vec3D;
+
+// Used to convert an unordered map (the type Inlet uses) to represent arrays
+// to a geometric vector type
+// FIXME: Clean this up/remove when PR #429 is merged
+Vector mapToVector(const std::unordered_map<int, double>& map)
+{
+  Vector result;
+  // Make sure the elements are accessed in order
+  for(int i = 0; i < 3; i++)
+  {
+    const auto ele = map.find(i);
+    if(ele != map.end())
+    {
+      result[i] = ele->second;
+    }
+  }
+  return result;
+}
+
+// A union of the members required for each of the operations is stored for simplicity
+struct Operator
+{
+  double rotate;
+  Vector axis;
+  Vector center;
+  Vector translate;
+  double x;
+  double y;
+  double z;
+  enum class Type
+  {
+    Translate,
+    Rotate,
+    Slice
+  } type;
+
+  // Again, the union of the necessary members are defined as part of the schema
+  static void defineSchema(inlet::Table& table)
+  {
+    table.addDouble("rotate");
+    // Vectors are defined as arrays of doubles
+    table.addDoubleArray("axis");
+    table.addDoubleArray("center");
+    table.addDoubleArray("translate");
+    // The slice operation can have sub-entries, so we represent it as a struct
+    // Note that Inlet does not require a 1-1 correspondence (or any correspondence)
+    // between structures defined in the schema via addStruct and structures extracted
+    // from the input file with FromInlet specializations (defined below)
+    auto& slice = table.addStruct("slice");
+    slice.addDouble("x");
+    slice.addDouble("y");
+    slice.addDouble("z");
+  }
+};
+
+template <>
+struct FromInlet<Operator>
+{
+  Operator operator()(const inlet::Table& base)
+  {
+    Operator result;
+    // Even though all the possible members are part of the schema, the
+    // "contains" operation can be used to check what is actually present
+    // in the input file
+    if(base.contains("translate"))
+    {
+      result.type = Operator::Type::Translate;
+      result.translate = mapToVector(base["translate"]);
+    }
+    else if(base.contains("rotate"))
+    {
+      result.type = Operator::Type::Rotate;
+      result.rotate = base["rotate"];
+      result.axis = mapToVector(base["axis"]);
+      result.center = mapToVector(base["center"]);
+    }
+    else if(base.contains("slice"))
+    {
+      result.type = Operator::Type::Slice;
+      // Grab the substructure corresponding to the slice operation
+      // and use it to populate the Operator instance
+      auto slice = base["slice"];
+      if(slice.contains("x"))
+      {
+        result.x = slice["x"];
+      }
+      if(slice.contains("y"))
```

Comment:
Minor: if `y` is not present in the input deck, 
* does/should it get initialized to some default value?
* how does the user know if `y` was present or if the default option was used?
* how/where does the developer add the `*.default()`  and/or `*.required()` tags?

Reply:
I think for an example this is indeed minor, but in actual usage it might be helpful to have a nullable utility type (like a limited version of `std::optional`) for these cases.

Regardless of how these slice variables are initialized, I don't think this example has any `default` or `required` fields - I think these would be useful to demonstrate.